### ISSUE TITLE
Vomnibar doesn't always respect dark mode

### ISSUE THREAD

**Issue by bobf32** [gdh1995/vimium-c#402]
**Describe the bug**
I would like the vomnibar to open in dark mode all the time, on all sites. Sometimes it does, sometimes not. I have a feeling this may have something to do with the effects of the Dark Reader extension which I also use.

**To Reproduce**
1. Go to URL https://en.wikipedia.org/ (Dark Reader is disabled for this site)
2. Click on o to open vomnibar
3. Vomnibar opens in dark mode as required.
4. Enable the Dark Reader extension for https://en.wikipedia.org/. The page re-renders in dark mode.
5. Click on o to open vomnibar
6. Vomnibar opens in light mode (when dark mode is desired).

I wonder if the vomnibar is trying to open in dark mode but dark reader is reversing the vomnibar colours to light, as in step 6. the sun picture, as opposed to moon, is shown in the top right of the vomnibar.

It will help a lot if here includes a screenshot when applicable.

**Browser, OS and Vimium C version**

4.1.2369.16 (Stable channel) (64-bit)

Edition	Windows 10 Pro
Version	21H1
Installed on	‎17/‎06/‎2020
OS build	19043.1165
Experience	Windows Feature Experience Pack 120.2212.3530.0

Vimium 1.90.2

If you're using MS Edge, Chrome, or other Chromium-based browsers,
include the browser and OS version found at `chrome://version`.
Also include the Vimium C version found at `chrome://extensions`.

If you're using Firefox, report the Firefox and OS version found at `about:support`.
Also include the Vimium C version found at `about:addons`.

**Comment by gdh1995** [comment]
Got. So your browser is "Vivaldi" on Windows, isn't it?

**Comment by bobf32** [comment]
Oops sorry, I thought I pasted that in. Yes, correct.

**Comment by gdh1995** [comment]
Yes `Dark Reader` inverts all color in the page (and inverts all images and videos again to keep them display normally).

It seems that there's no valid way to escape such color filters. But I can add special logic to make Vomnibar just work with the `Dark Reader` itself.

**Comment by bobf32** [comment]
Thanks, that would be great. I thought I'd add some more info to expand on my particular use case. I try and configure my browser so that all pages render in dark mode. To do this for sites that render only in light mode I configure Dark Reader to invert the colours. For sites that render in dark mode (or allow this as an option) I disable Dark Reader and let the site render in dark mode as it does natively. 

However I would like the vomnibar to show in dark mode all the time, regardless of whether Dark Reader is affecting the site. 

Hope that helps.

**Comment by gdh1995** [comment]
Then you may try `static` and `dynamic` mode of Dark Reader: these modes won't apply a "global" filter so Vomnibar should be in dark again.

![image](https://user-images.githubusercontent.com/5547703/129353891-10947654-00e9-4dfc-9511-0cd368912816.png)

**Comment by bobf32** [comment]
Thanks. For some sites this is acceptable, but not all.

**Comment by gdh1995** [comment]
Hello, please try this developing version:  [vimium_c-1.92.0-chrome-7f407f6-mod.zip](https://github.com/gdh1995/vimium-c/files/6982563/vimium_c-1.92.0-chrome-7f407f6-mod.zip). Usage: https://github.com/gdh1995/vimium-c/wiki/Test-a-zip-package

**Comment by bobf32** [comment]
Should I uninstall the previous version first?

**Comment by gdh1995** [comment]
&nbsp;Itshould auto override



---Original---
From: ***@***.***&gt;
Date: Fri, Aug 13, 2021 21:14 PM
To: ***@***.***&gt;;
Cc: "Dahan ***@***.******@***.***&gt;;
Subject: Re: [gdh1995/vimium-c] Vomnibar doesn't always respect dark mode (#402)




 
Should I uninstall the previous version first?
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by bobf32** [comment]
That looks great. Vomnibar now renders in dark mode on pages whether or not Dark Reader is active. Works with all four modes of Dark Reader too.

**Comment by bobf32** [comment]
I notice an error in the extension page, see attached pics.

![2021-08-13 14_38_07-Extensions - Vivaldi](https://user-images.githubusercontent.com/6544564/129365710-6f353564-a0b4-4b28-8b48-f50c44cdfe9d.png)

![2021-08-13 14_38_15-Extensions - Vivaldi](https://user-images.githubusercontent.com/6544564/129365709-fcc1d7aa-f368-41b8-a04a-7edd28cfacf1.png)

**Comment by gdh1995** [comment]
It doesn't matter. It's used to work with the `#extensions-on-chrome-urls` on `chrome://flags/` (#239).

**Comment by gdh1995** [comment]
v1.92.1 has been published on Chrome and Firefox stores, and you may take a try again.
